tests/js/client/shell/shell-auto-refill-index-caches-on-modify-noncluster.js

```diff
@@ -31,15 +31,45 @@ const getMetric = require('@arangodb/test-helper').getMetricSingle;
 const cn = 'UnitTestsCollection';
 const n = 5000; // documents
 
-const waitForMetrics = () => {
-  // wait for metrics to settle, using an informal API
+const waitForPendingRefills = () => {
+  // wait for pending refill operations to have finished, using an informal API
   arango.POST('/_api/index/sync-caches', {});
 };
 
+const runWithRetry = (cb) => {
```

Comment:
This function should be put to test-helper.js

Reply:
https://github.com/arangodb/release-test-automation/blob/main/test_data/makedata.js#L139 already has a similar function.

Reply:
This `runWithRetry` function does something specific to a collection used in the test, e.g.
```
      if (tries++ === 3) {
        throw err;
      }   

      db[cn].truncate();
      require("internal").sleep(2);
```
In order to make this function generic, I would need to add a callback function for the failure case, make the sleep value configurable and also the number of tries. This would add 3 more parameters to the generic function.
```
const runWithRetry = (cb, failureCb, tries, sleepTimeout) => { ... }
```
I would also have to adjust all callers of this function.

Not sure if that's really worth it. I doubt it.

Reply:
Y? 
If s.b. needs it, he can adapt it to get more functionality? 
I have cleaned up so many of these "just need them here" helper functions through out the tests in the last 24 months, we definitely should stop that now.

Reply:
done